mlos_core/mlos_core/optimizers/flaml_optimizer.py

```diff
@@ -99,14 +101,15 @@ def _register(self, configurations: pd.DataFrame, scores: pd.DataFrame,
         """
         if context is not None:
             warn(f"Not Implemented: Ignoring context {list(context.columns)}", UserWarning)
-        for (_, config), score in zip(configurations.astype('O').iterrows(),
-                                      scores[self._flaml_optimization_target]):
+        for (_, config), (_, score) in zip(configurations.astype('O').iterrows(), scores.iterrows()):
             cs_config: ConfigSpace.Configuration = ConfigSpace.Configuration(
                 self.optimizer_parameter_space, values=config.to_dict())
             if cs_config in self.evaluated_samples:
                 warn(f"Configuration {config} was already registered", UserWarning)
-
-            self.evaluated_samples[cs_config] = EvaluatedSample(config=config.to_dict(), score=score)
+            self.evaluated_samples[cs_config] = EvaluatedSample(
+                config=config.to_dict(),
+                score=float(np.average(score.astype(float), weights=self._objective_weights)),
+            )
```

Comment:
This is technically also adding a basic version of multi-objective support for FLAML, correct?
Can you add that to the PR description/title?  Seems like an important feature to make note of.

Reply:
done!